sdk/core/azure-core/test/ut/curl_session_test.cpp

```diff
@@ -82,6 +82,130 @@ namespace Azure { namespace Core { namespace Test {
     Azure::Core::Http::CurlConnectionPool::ConnectionPoolIndex.clear();
   }
 
+  TEST_F(CurlSession, chunkBadFormatResponse)
+  {
+    // chunked response with unexpected char at the end
+    std::string response("HTTP/1.1 200 Ok\r\ntransfer-encoding: chunked\r\n\r\n9\r\n");
+    std::string response2("123456789\r\n0\r\n\rx");
+    std::string connectionKey("connection-key");
+
+    // Can't mock the curMock directly from a unique ptr, heap allocate it first and then make a
+    // unique ptr for it
+    MockCurlNetworkConnection* curlMock = new MockCurlNetworkConnection();
+    EXPECT_CALL(*curlMock, SendBuffer(_, _, _)).WillOnce(Return(CURLE_OK));
+    EXPECT_CALL(*curlMock, ReadFromSocket(_, _, _))
+        .WillOnce(DoAll(
+            SetArrayArgument<1>(response.data(), response.data() + response.size()),
+            Return(response.size())))
+        .WillOnce(DoAll(
+            SetArrayArgument<1>(response2.data(), response2.data() + response2.size()),
+            Return(response2.size())));
+    EXPECT_CALL(*curlMock, GetConnectionKey()).WillRepeatedly(ReturnRef(connectionKey));
+    EXPECT_CALL(*curlMock, updateLastUsageTime());
+    EXPECT_CALL(*curlMock, DestructObj());
+
+    // Create the unique ptr to take care about memory free at the end
+    std::unique_ptr<MockCurlNetworkConnection> uniqueCurlMock(curlMock);
+
+    // Simulate a request to be sent
+    Azure::Core::Http::Url url("http://microsoft.com");
+    Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Get, url);
+
+    {
+      // Create the session inside scope so it is released and the connection is moved to the pool
+      auto session = std::make_unique<Azure::Core::Http::CurlSession>(
+          request, std::move(uniqueCurlMock), true);
+
+      EXPECT_NO_THROW(session->Perform(Azure::Core::GetApplicationContext()));
+      auto response = session->GetResponse();
+      response->SetBodyStream(std::move(session));
+      auto bodyS = response->GetBodyStream();
+
+      // Read the bodyStream to get get all chunks
+      EXPECT_THROW(
+          Azure::Core::Http::BodyStream::ReadToEnd(Azure::Core::GetApplicationContext(), *bodyS),
+          Azure::Core::Http::TransportException);
+    }
+    // Clear the connections from the pool to invoke clean routine
+    Azure::Core::Http::CurlConnectionPool::ConnectionPoolIndex.clear();
+  }
+
+  TEST_F(CurlSession, chunkSegmentedResponse)
```

Comment:
Any way to incorporate parts of this test to be general purpose and get some more winhttp transport coverage as well?

Reply:
We do manual parsing of a chunked response only with libcurl.  For winHttp, we delegate the job to winHttp (which is perfectly splendid! :)